api/src/main/java/io/cloudevents/CloudEvent.java

```diff
@@ -15,64 +15,32 @@
  */
 package io.cloudevents;
 
-import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
-import io.cloudevents.impl.DefaultCloudEventImpl;
-
-import java.net.URI;
-import java.time.ZonedDateTime;
-import java.util.List;
+import java.util.Map;
 import java.util.Optional;
 
 /**
- * An abstract event envelope, representing the 0.2 version of the <a href="https://github.com/cloudevents/spec/blob/master/spec.md">CNCF CloudEvent spec</a>.
+ * An abstract event envelope
+ * 
+ * @author fabiojose
  *
+ * @param <A> The attributes type
+ * @param <T> The 'data' type
  */
-@JsonDeserialize(as = DefaultCloudEventImpl.class)
-public interface CloudEvent<T> {
-
-    /**
-     * Type of occurrence which has happened. Often this property is used for routing, observability, policy enforcement, etc.
-     */
-    String getType();
-
-    /**
-     * The version of the CloudEvents specification which the event uses. This enables the interpretation of the context.
-     */
-    String getSpecVersion();
-
-    /**
-     * This describes the event producer. Often this will include information such as the type of the event source, the organization publishing the event, and some unique identifiers.
-     * The exact syntax and semantics behind the data encoded in the URI is event producer defined.
-     */
-    URI getSource();
-
-    /**
-     * ID of the event. The semantics of this string are explicitly undefined to ease the implementation of producers. Enables deduplication.
-     */
-    String getId();
-
-    /**
-     * Timestamp of when the event happened.
-     */
-    Optional<ZonedDateTime> getTime();
-
-    /**
-     * A link to the schema that the data attribute adheres to.
-     */
-    Optional<URI> getSchemaURL();
-
-    /**
-     * Describe the data encoding format
-     */
-    Optional<String> getContentType();
-
-    /**
-     * The event payload. The payload depends on the eventType, schemaURL and eventTypeVersion, the payload is encoded into a media format which is specified by the contentType attribute (e.g. application/json).
-     */
-    Optional<T> getData();
-
-    /**
-     *
-     */
-    Optional<List<Extension>> getExtensions();
-}
\ No newline at end of file
+public interface CloudEvent<A extends Attributes, T> {
```

Comment:
Won't this change break existing users significantly? I suppose compatibility is no concern at this point?